thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -80,7 +85,7 @@ public DataAvailabilityTaskScheduler(long delayInSec, long fallBackTimeInSec) {
   @Override
   public void run() {
     Map<DetectionConfigDTO, Set<String>> detection2DatasetMap = new HashMap<>();
-    Map<String, Long> dataset2RefreshTimeMap = new HashMap<>();
+    Map<String, Pair<Long, Long>> dataset2RefreshTimeMap = new HashMap<>();
```

Comment:
Using Pair<Long, Long> for <last refresh time, last refresh event time> may make the code a little hard to read without context. Would it be better to create a class for this instead?

Reply:
+1
Let's use a class since it is complex enough now.